courses/C01-package_express_fusen.en.Rmd

```diff
@@ -0,0 +1,540 @@
+
+- Panel "Build" > "Check"
+    + or `devtools::check()`  
+  
+- Solve potential problems in the "dev_history.Rmd"
+- Re-inflate the package if necessary
+
+- Reach **0 Error, 0 Warnings, 0 Notes**
+
+<br/>
+
+> Note that `fusen::inflate()` already launch this two commands
+
+???
+
+It is always good to know this commands, although `inflate()` already do it. You may need them if you go back to a classical way of maintaining your package
+
+---
+class: slide 
+
+### Step 9 : Install and use your package
+
+- Restart your RStudio session
+  + The 'Build' tab should appear in RStudio
+  
+- Install the package
+  + Panel "Build" > "Install and Restart
+  
+- Test the package directly in the console
+  + `hello::say_hello("Toto")`
+
+- Test the knit of the thumbnail
+  
+- Check that the help for your function appears
+  + `?say_hello`
+
+---
+class: slide
+
+### Step 10 : Present the documentation
+
+Creation of a web site dedicated to the presentation of the package
+
+- Add these lines in the `development` chunk and run
+
+```{r, eval=FALSE}
+usethis::use_pkgdown()
+pkgdown::build_site()
+```
+
+```{r echo=FALSE, out.width="70%"}
+knitr::include_graphics("images/squirrels_pkgdown.png")
+```
+
+
+???
+
+usethis::use_pkgdown() will put the /docs folder in the gitignore, and the buildignore.
+pkgdown::build_site() will drop the site into the docs folder by default
+
+---
+class: slide
+
+### Your turn
+
+- Second round : accompanied driving
+- Third round : in autonomy
+
+> Edit "dev_history.Rmd" and re-execute `inflate()` as many times as necessary until everything runs smoothly.
+
+```{r echo=FALSE, out.width="90%"}
+knitr::include_graphics("images/fusen_rmd_folds_pkg.png")
+```
+
+---
+class: slide
+
+### Quiz
+
+Where were the pieces of code for the :
+
+- `description` ?
+- `function` ?
+- `example` ?
+- `tests` ?
+- `development` ?
+
+--
+
+Let's check it together...
+
+- You can run an example of your function directly with Ctrl + Enter
+- You can run the unit test by clicking on "Run test"
+- You can run the vignette if your package is installed
+
+But be careful, if you want to modify some code, go back to the "dev_history.Rmd" and do `inflate()` again
+
+???
+
+- `description` : in DESCRIPTION
+- `function` : in the independent .R file with the name of the function
+- `example` : in the independent .R file
+  + in the independent .R file with the function name
+  + in the thumbnail
+- tests`: in the independent tests/testthat folder with the function name
+- development`: nowhere. it stays in the development tracking file "dev_history.Rmd
+
+
+
+---
+class: slide 
+
+### The rest is just as important
+
+- Doing _version control_ with _git
+  + `usethis::use_git()`
+  
+- Add the README to your git repository
+  + `usethis::use_readme_rmd()`
+  
+- Use a test set for your examples and tests
+  + `usethis::use_data()`: function to create an internal dataset  
+  
+- Add a version tracking file
```

Comment:
Should we also mention `usethis::use_version()`? Participants coming from an analysis context might also appreciate advise on when/how often to update different parts of the version number

Reply:
That is a common question yes. Let's add this.